draft-reddy-pquip-pqc-hsm.md

```diff
@@ -390,9 +387,12 @@ provide strong security assurances over extended periods. ML-DSA is not included
 list because it is a lattice-based signature scheme, making it susceptible to potential
 advances in quantum and classical attacks on structured lattices. The long-term security
 of ML-DSA depends on the continued hardness of lattice-based problems, which remain an
-active area of research. In contrast, SLH-DSA, HSS-LMS, and XMSS are based on well-studied
+active area of research. Similarly, although SLH-DSA (Stateless Hash-Based Digital Signature Algorithm),
+a stateless hash-based signature scheme, avoids the complexity of state management, its
+relatively large signature sizes make it less practical for constrained devices with memory
+limitations. In contrast, HSS-LMS and XMSS are based on well-studied
 hash functions, ensuring their security does not rely on unproven assumptions about
-lattice hardness. Given this uncertainty, use of a hash-based signature such as SLH-DSA
+lattice hardness. Given this uncertainty, use of a hash-based signature
```

Comment:
How about 


"Since ML-DSA implementations are still maturing, hash-based signatures could be a more reliable choice for firmware authentication, especially when long-term cryptographic stability is important."

Reply:
I don't consider signature size of *firmware* to be much of a constraint; I mean, the firmware itself is likely to be 100s of KB.  The SLH-DSA signature is 17KB.  Meh.  Call it a 20% tax if firmware is 100KB.  During firmware upgrade, the HSM is not doing its usual cryptographic operations -- it is solely doing its firmware upgrade.